tools/azure-sdk-tools/tests/conftest.py

```diff
@@ -0,0 +1,25 @@
+import pytest
+import tempfile
+import os
+
+from typing import List
+from tempfile import TemporaryDirectory
+
+def create_temp_directory(fake_creation_paths: List[str]) -> TemporaryDirectory:
+    tmp_dir = TemporaryDirectory()
+
+    for file in fake_creation_paths:
+        target_path = os.path.join(tmp_dir.name, file)
+        dirname = os.path.join(tmp_dir.name, os.path.dirname(file))
+
+        if not os.path.exists(dirname):
+            os.mkdir(dirname)
+
+        with open(target_path, "w"):
+            pass
+
+    return tmp_dir
+
+@pytest.fixture()
+def tmp_directory_create():
+    return create_temp_directory
```

Comment:
nitpick: TemporaryDirectory can be used as a context manager to handle cleanup.

```suggestion
@pytest.fixture()
def tmp_directory_create():
    with TemporaryDirectory() as tmp_dir:
        def create_temp_directory(fake_creation_paths: List[str]) -> TemporaryDirectory:
            for file in fake_creation_paths:
                target_path = os.path.join(tmp_dir.name, file)
                dirname = os.path.join(tmp_dir.name, os.path.dirname(file))

                if not os.path.exists(dirname):
                    os.mkdir(dirname)

                with open(target_path, "w"):
                    pass
            return tmp_dir

        yield create_temp_directory
```

Reply:
also, small style fyi that doesn't need to be addressed:

Pathlib makes a lot of that path manipulation more concise (and clearer?).

```python
from pathlib import Path

@pytest.fixture()
def tmp_directory_create():
    with TemporaryDirectory() as tmp_dir:
        def create_temp_directory(fake_creation_paths: List[str]) -> TemporaryDirectory:
            for file in fake_creation_paths:
                target_path = Path(tmp_dir.name, file)
                target_path.parent.mkdir(parents=True, exist_ok=True)
                target_path.touch()

            return tmp_dir

        yield create_temp_directory

```

Reply:
Using a closure to get the TempDirectory in scope is brilliant. Definitely wasn't thinking like this at all but I really like the end result. Thanks for the suggestion!